pkg/scheduler/algorithm/priorities/metadata.go

```diff
@@ -113,3 +114,14 @@ func getSelectors(pod *v1.Pod, sl algorithm.ServiceLister, cl algorithm.Controll
 
 	return selectors
 }
+
+func getNonZeroRequests(pod *v1.Pod) *schedulernodeinfo.Resource {
+	result := &schedulernodeinfo.Resource{}
+	for i := range pod.Spec.Containers {
```

Comment:
why does this not consider init containers ?

Reply:
I have just moved this existing method from one file to another, it was defined in the resource_allocation code base which i changed so it no more uses it, but after deleting it I realized it's being called by other methods outside the scope as well so just redefined it in that class. Did not want to break something. Also, my assumption is it does not consider init containers coz it is calculating the resources on containers which are already running.